streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -750,7 +750,7 @@ protected void maybeClean() {
         long now = time.milliseconds();
 
         if (now > lastCleanMs + cleanTimeMs) {
```

Comment:
We only want to run the check on the interval. And then when it does run we only want to consider directories where the last mod is at least `cleanTimeMs` ago.